src/arena.c

```diff
@@ -1080,8 +1088,10 @@ arena_cache_bin_fill_small(tsdn_t *tsdn, arena_t *arena,
 	unsigned binshard;
 	bin_t *bin = arena_bin_choose(tsdn, arena, binind, &binshard);
 
+	arena_bin_flush_batch_state_t batch_flush_state;
 label_refill:
 	malloc_mutex_lock(tsdn, &bin->lock);
+	arena_bin_flush_batch_after_lock(tsdn, arena, bin, binind, &batch_flush_state);
```

Comment:
It seems this happens only at refill. It feels to me that we should also try this at tcache flush (flushing the batched of the home bin)?

Reply:
We do this -- in `tcache_bin_flush_impl_small`, in the `arena_dalloc_bin_locked` sequence.